src/common/libraries/MessageLib.sol

```diff
@@ -1086,7 +1086,7 @@ library MessageLib {
         });
     }
 
-    function serialize(UpdateHoldingAmount memory t) public pure returns (bytes memory) {
+    function serialize(UpdateHoldingAmount memory t) internal pure returns (bytes memory) {
```

Comment:
This is really important: libraries should only have internal methods. Otherwise they become externally referenced libraries, which are effectively separate contract deployments. These have all sorts of issues and should not be used.

Reply:
I think it was me frightening with the annoying deep stack issue that happened when serializing that message. I wanted to force an external call there. Thanks for the eagle eye!